Java/factorial.java

```diff
@@ -0,0 +1,20 @@
+import java.util.Scanner;
+public class JavaExample {
+
+    public static void main(String[] args) {
+
+        int number;
+        System.out.println("Enter the number: ");
+        Scanner scanner = new Scanner(System.in);
+        number = scanner.nextInt();
+        scanner.close();
+        long fact = 1;
+        int i = 1;
+        while(i<=number)
+        {
+            fact = fact * i;
+            i++;
+        }
+        System.out.println("Factorial of "+number+" is: "+fact);
+    }
```

Comment:
Always use .close() method before exiting the program for a Scanner object to prevent memory leakage.